app/models/phone_number_authentication/authenticate.rb

```diff
@@ -40,19 +44,42 @@ def verify_auth
       case
       when authentication.nil?
         failure('login.error_messages.unknown_user')
+      when authentication.locked_at
+        if authentication.locked_at >= LOCKOUT_TIME.ago
+          minutes_left = (LOCKOUT_TIME - (Time.current - authentication.locked_at)) / 1.minute
+          minutes_left = minutes_left.round
+          failure("login.error_messages.account_locked", minutes: minutes_left)
+        else
+          unlock
+          verify_auth
+        end
       when authentication.otp != otp
+        track_failed_attempt
         failure('login.error_messages.invalid_otp')
       when !authentication.otp_valid?
+        track_failed_attempt
         failure('login.error_messages.expired_otp')
       when !authentication.authenticate(password)
+        track_failed_attempt
         failure('login.error_messages.invalid_password')
       else
         success
       end
     end
 
-    def failure(message_key)
-      error_string = I18n.t(message_key)
+    def unlock
+      authentication.update!(locked_at: nil, failed_attempts: 0)
```

Comment:
Could we add an `#unlock` method to the PhoneNumberAuthentication model? I figure this would be a good way for us to cleanly unlock an account if we need to do so manually.